### PR TITLE
Added chat test mock to imitate chat users.

### LINKED ISSUE #36: Create AI Chatters to play against
Create AI chatters. This would enable the following

- Stress testing
- Feature testing
- Demoing
- Practicing offline

### PR DESCRIPTION
**Pull request by ootsby**
This generates "bot" users to facilitate testing. Testing without a chat channel or with few active users in chat or if you want to test a flood of commands etc.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -4,15 +4,20 @@ use ggez::{event, ContextBuilder};
...
 use twitch_chat_wrapper::chat_message::ChatMessage;
+mod chat_test_mock;
 
 const WINDOW_SIZE: (f32, f32) = (1920.0, 1080.0);
 
 fn main() {
     let (send_to_game, receive_from_twitch) = channel::<ChatMessage>();
     let (send_to_twitch, receive_from_game) = channel::<String>();
+    
+    chat_test_mock::run(send_to_game.clone(), 5, get_the_streamer_game::SPLASH_DURATION, 500, 5000);
```
Comment (LordMZTE) [inline src/main.rs:15]: this should definetely go into a test function, ideally also in `chat_test_mock.rs`

File: src/chat_test_mock.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+use rand::Rng;
+use std::sync::mpsc::Sender;
+use std::thread;
+use std::time::Duration;
+use twitch_chat_wrapper::chat_message::ChatMessage;
+use twitchchat::messages::Privmsg;
+use twitchchat::FromIrcMessage;
+
+pub fn run(
```
Comment (LordMZTE) [inline src/chat_test_mock.rs:9]: this should go into the `test` folder, as this is an integration test

**Review by LordMZTE — CHANGES_REQUESTED** [review]

**Comment by BrooksPatton** [comment]
This is a great idea, I've created an issue (#36) to track this along with a branch ( 36-create-ai-chatters-to-play-against ) to work on it. As mentioned in chat I'll be happy to make the architectural changes that I put into the issue. If you (or anyone else) would like to contribute let me know here and we'll figure a way to do so.

**Comment by LordMZTE** [comment]
> 
> 
> This is a great idea, I've created an issue (#36) to track this along with a branch ( 36-create-ai-chatters-to-play-against ) to work on it. As mentioned in chat I'll be happy to make the architectural changes that I put into the issue. If you (or anyone else) would like to contribute let me know here and we'll figure a way to do so.

this is not a bot! this is purely for testing purposes, which is why the invocation of the mock function in main is completely incorrect

**Pull request closed without merging**